client/app/components/dashboards/dashboard-widget/VisualizationWidget.jsx

```diff
@@ -161,16 +162,19 @@ function VisualizationWidgetFooter({ widget, isPublic, onRefresh, onExpand }) {
             className="refresh-button hidden-print btn btn-sm btn-default btn-transparent"
             onClick={() => refreshWidget(1)}
             data-test="RefreshButton">
-            <i className={cx("zmdi zmdi-refresh", { "zmdi-hc-spin": refreshClickButtonId === 1 })} />{" "}
+            <i className={cx("zmdi zmdi-refresh", { "zmdi-hc-spin": refreshClickButtonId === 1 })} aria-hidden="true" />
+            <span className="sr-only">
+              {refreshClickButtonId === 1 ? "Refreshing, please wait. " : "Press to refresh. "}
```

Comment:
Do or not: feels like some duplication, here -- if you feel like you can abstract it, go for it! But if you think it could create business logic changes, don't worry about  it.